programs/wen_transfer_guard/src/instructions/initialize.rs

```diff
@@ -0,0 +1,58 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{token_2022::spl_token_2022::ID as TOKEN_2022_PROGRAM_ID, token_interface::Mint};
+use spl_tlv_account_resolution::{account::ExtraAccountMeta, state::ExtraAccountMetaList};
+use spl_transfer_hook_interface::{error::TransferHookError, instruction::ExecuteInstruction};
+
+use crate::{AnchorExtraAccountMeta, EXTRA_ACCOUNT_METAS};
+
+#[derive(Accounts)]
+#[instruction(metas: Vec<AnchorExtraAccountMeta>)]
+pub struct Initialize<'info> {
+    /// CHECK: This account's data is a buffer of TLV data
+    #[account(
+        init,
+        space = ExtraAccountMetaList::size_of(metas.len()).unwrap(),
```

Comment:
Should use a result and throw a structured error.
```suggestion
        space = ExtraAccountMetaList::size_of(metas.len())? @ ErrorCode::MismatchMetaListSize,
```

Reply:
I don't think this is valid anchor lang syntax for spacing :(